src/silo/preprocessing/sequence_info.cpp

```diff
@@ -79,12 +79,46 @@ std::string SequenceInfo::getAminoAcidSequenceSelect(
    );
 }
 
+void validateStruct(
+   std::vector<std::string> names_to_validate,
+   std::vector<std::string> names_to_validate_against,
+   std::string name_type,
+   const std::filesystem::path& input_filename
+) {
+   for (const std::string& name : names_to_validate) {
+      if (std::find(names_to_validate_against.begin(), names_to_validate_against.end(), name) ==
+          names_to_validate_against.end()) {
+         throw silo::preprocessing::PreprocessingException(fmt::format(
+            "The {} {} which is contained in the input file {} is "
+            "not contained in the reference sequences.",
+            name_type,
+            name,
+            input_filename.string()
+         ));
+      }
+   }
+   for (const std::string& name : names_to_validate_against) {
+      if (std::find(names_to_validate.begin(), names_to_validate.end(), name) == names_to_validate.end()) {
+         // TODO(#220) handle the cases when segments are left out appropriately
```

Comment:
Is this still a TODO? #220 is already done.

Reply:
I was also under the impression that this already works, but didn't want to remove the TODO without verifying everything first

Reply:
Then let's make it a TODO for #129?

Reply:
I would add another ticket that request that keys may be left out
But might turn into a won't fix (/ won't implement), as duckdb validation might not be able to cover left out keys in all cases